app/src/main/java/org/piwigo/helper/URLHelper.java

```diff
@@ -0,0 +1,79 @@
+package org.piwigo.helper;
+
+import android.app.Activity;
+import android.app.ProgressDialog;
+import android.os.AsyncTask;
+
+import org.piwigo.ui.login.LoginActivity;
+import org.piwigo.ui.main.MainActivity;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.UnknownHostException;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLException;
+import javax.net.ssl.SSLHandshakeException;
+import javax.net.ssl.SSLPeerUnverifiedException;
+
+public class URLHelper extends AsyncTask<String, Void, String> {
+
+    @Override
+    protected String doInBackground(String... url) {
+        url[0] = url[0].replaceAll("https://", "").replaceAll("http://", "");
```

Comment:
This means you force the user to https even if he gave http and http would work...
I agree that we should recommend the users to switch to https if that works, but we should not force him.